ouroboros/shared/value.go

```diff
@@ -54,19 +56,53 @@ func Subtract(a Value, b Value) Value {
 
 func Enough(have Value, want Value) (bool, error) {
 	for policyId, assets := range want {
-		for assetName, amt := range assets {
-			if haveAssets, ok := have[policyId]; ok {
-				if haveAssets[assetName].Int64() < amt.Int64() {
-					return false, fmt.Errorf("not enough %v.%v to meet demand", policyId, assetName)
+		if haveAssets, ok := have[policyId]; ok {
+			for assetName, amt := range assets {
+				if haveAssets[assetName].BigInt().Cmp(amt.BigInt()) == -1 {
```

Comment:
Actually, let me put this back the way it was. One sec.

Reply:
Actually, unless there's a reason for casting to `Int64`, I have a soft preference for keeping this and all the other `Cmp()` calls as they are, even if they're a bit clunky. I'm not sure I see a point for casting, other than making the line a bit cleaner.

Reply:
I would add `LessThan` and `GreaterThan` method to our little num library; the whole point is to never have to interact with big.Int directly, because it behaves really unintuitively and has lead to many bugs.